src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/flow/resources/FlowResourcesManager.java

```diff
@@ -147,8 +172,58 @@ FlowResources allocateResources(Flow flow, PathId forwardPathId, PathId reverseP
                 .build();
     }
 
-    private EncapsulationResourcesProvider getEncapsulationResourcesProvider(FlowEncapsulationType type) {
-        EncapsulationResourcesProvider provider = encapsulationResourcesProviders.get(type);
+    @VisibleForTesting
+    HaFlowResources allocateResources(
+            HaFlow haFlow, PathId forwardPathId, PathId reversePathId, SwitchId yPointSwitchId)
+            throws ResourceAllocationException {
+        HaPathResources.HaPathResourcesBuilder forward = HaPathResources.builder()
+                .pathId(forwardPathId);
+        HaPathResources.HaPathResourcesBuilder reverse = HaPathResources.builder()
+                .pathId(reversePathId);
+        Map<String, PathId> reverseSubPathIds = new HashMap<>();
+        for (int i = 0; i < haFlow.getSubFlows().size(); i++) {
+            char suffix = (char) (HA_SUB_PATH_BASE_SUFFIX + i);
```

Comment:
yeah, I understand, but I was pointing at possible high value of i. Even though 30 subflows sound unrealistic, I thought it is better to throw an exception in this case rather than having the prefix `DEL` symbol

Reply:
Added check